### PR TITLE
add `GetRootTopics` client function

### PR DESCRIPTION
**Pull request by justinpjose**
### What
**Describe what you have changed and why.**
- Organise models package
- Separate private and public handlers for `topics` in different files
- Upgrade dependencies
- Add `GetRootTopics` client func for web and publishing

### How to review
**Describe the steps required to test the changes.**
- Check if the code changes make sense
- Check if the test passes

### Who can review
**Describe who worked on the changes, so that other people can review.**
- Anyone

### CODE REVIEW

File: api/topic_private_test.go [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+	"github.com/ONSdigital/dp-topic-api/apierrors"
+	"github.com/ONSdigital/dp-topic-api/config"
+	"github.com/ONSdigital/dp-topic-api/models"
+	storeMock "github.com/ONSdigital/dp-topic-api/store/datastoretest"
+
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestGetTopicPrivateHandler(t *testing.T) {
+
```
Comment (rahulPalliyalil) [inline api/topic_private_test.go:21]: (suggestion): I can you can run these tests in parallel ? ```t.Parallel``` ?

File: api/topic_private_test.go [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+					retTopic := models.TopicResponse{}
+					err = json.Unmarshal(payload, &retTopic)
+					So(err, ShouldBeNil)
+					So(retTopic.ID, ShouldEqual, "1")
+					So(retTopic.Next.ID, ShouldEqual, "1")
+				})
+			})
+
+			// No more tests needed because getting the 'topic_root' makes use of
+			// function getSubtopicsPublicByID() which is fully tested in
```
Comment (rahulPalliyalil) [inline api/topic_private_test.go:268]: (suggestion): Not sure if you need this comment ;)

File: models/navigation.go [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package models
+
+type Navigation struct {
```
Comment (rahulPalliyalil) [inline models/navigation.go:3]: (suggestion): Probably worth adding a comment on what ```Navigation``` does for its consumers to understand as this is an exported api.

**Review by rahulPalliyalil — DISMISSED** [review]

File: models/navigation.go (hunk 3 above)
Comment (justinpjose) [reply models/navigation.go:4]: Done

File: api/topic_private_test.go (hunk 1 above)
Comment (justinpjose) [reply api/topic_private_test.go:22]: Done

**Pull request merged**